tests/MyGame/Example/Monster.java

```diff
@@ -344,5 +347,372 @@ public static Monster __lookup_by_key(Monster obj, int vectorLocation, String ke
+    switch (_oAnyAmbiguousType) {
+      case AnyAmbiguousAliases.M1:
+        _oAnyAmbiguous.setValue(_oAnyAmbiguousValue != null ? ((MyGame.Example.Monster) _oAnyAmbiguousValue).unpack() : null);
+        break;
+      case AnyAmbiguousAliases.M2:
+        _oAnyAmbiguous.setValue(_oAnyAmbiguousValue != null ? ((MyGame.Example.Monster) _oAnyAmbiguousValue).unpack() : null);
+        break;
+      case AnyAmbiguousAliases.M3:
+        _oAnyAmbiguous.setValue(_oAnyAmbiguousValue != null ? ((MyGame.Example.Monster) _oAnyAmbiguousValue).unpack() : null);
+        break;
+      default: break;
+    }
+    _o.setAnyAmbiguous(_oAnyAmbiguous);
+    List<Integer> _oVectorOfEnums = new ArrayList<>();
+    for (int _j = 0; _j < this.vectorOfEnumsLength(); ++_j) {_oVectorOfEnums.add(this.vectorOfEnums(_j));}
+    _o.setVectorOfEnums(_oVectorOfEnums);
+    byte _oSignedEnum = this.signedEnum();
+    _o.setSignedEnum(_oSignedEnum);
+    List<Integer> _oTestrequirednestedflatbuffer = new ArrayList<>();
+    for (int _j = 0; _j < this.testrequirednestedflatbufferLength(); ++_j) {_oTestrequirednestedflatbuffer.add(this.testrequirednestedflatbuffer(_j));}
+    _o.setTestrequirednestedflatbuffer(_oTestrequirednestedflatbuffer);
+    List<MyGame.Example.StatT> _oScalarKeySortedTables = new ArrayList<>();
+    for (int _j = 0; _j < this.scalarKeySortedTablesLength(); ++_j) {_oScalarKeySortedTables.add(this.scalarKeySortedTables(_j) != null ? this.scalarKeySortedTables(_j).unpack() : null);}
+    _o.setScalarKeySortedTables(_oScalarKeySortedTables);
+  }
+  public static int pack(FlatBufferBuilder builder, MonsterT _o) {
+    if (_o == null) return 0;
+    int _name = _o.getName() == null ? 0 : builder.createString(_o.getName());
+    int _inventory = 0;
+    if (_o.getInventory() != null) {
+      int[] __inventory = new int[_o.getInventory().size()];
+      int _j = 0;
+      for (int _e : _o.getInventory()) { __inventory[_j] = _e; _j++;}
+      _inventory = createInventoryVector(builder, __inventory);
+    }
+    byte _testType = _o.getTest() == null ? MyGame.Example.Any.NONE : _o.getTest().getType();
+    int _test = _o.getTest() == null ? 0 : AnyUnion.pack(builder, _o.getTest());
+    int _test4 = 0;
+    if (_o.getTest4() != null) {
+      startTest4Vector(builder, _o.getTest4().size());
+      for (MyGame.Example.TestT _e : _o.getTest4()) { MyGame.Example.Test.pack(builder, _e);}
+      _test4 = builder.endVector();
+    }
+    int _testarrayofstring = 0;
+    if (_o.getTestarrayofstring() != null) {
+      int[] __testarrayofstring = new int[_o.getTestarrayofstring().size()];
+      int _j = 0;
+      for (String _e : _o.getTestarrayofstring()) { __testarrayofstring[_j] = builder.createString(_e); _j++;}
+      _testarrayofstring = createTestarrayofstringVector(builder, __testarrayofstring);
+    }
+    int _testarrayoftables = 0;
+    if (_o.getTestarrayoftables() != null) {
+      int[] __testarrayoftables = new int[_o.getTestarrayoftables().size()];
```

Comment:
This temp `int[]` is going to be hard to avoid. We could consider keep a single stack of offsets, but for now its probably easier to leave this as-is.